src/EoS/Nuclear/findtoreps_direct.cpp

```diff
@@ -0,0 +1,191 @@
+// Description :  Bilinear interpolation
+//
+// Note        :  1. Invoked by findtoreps_direct()
+//
+// Return      :  interpolated value
+//-------------------------------------------------------------------------------------
+GPU_DEVICE
+real Bilinear( const real *table, const real frac_x, const real frac_y,
+               const int ifs, const int nxy )
+{
+
+   const real left  = table[ifs      ] + ( table[ifs + 1      ] - table[ifs      ] ) * frac_x;
+   const real right = table[ifs + nxy] + ( table[ifs + 1 + nxy] - table[ifs + nxy] ) * frac_x;
+
+   return left + ( right - left ) * frac_y;
+
+} // FUNCTION : Bilinear
+
+
+
+//-------------------------------------------------------------------------------------
+// Function    :  findtoreps_direct
+// Description :  Finding temperature of internal energy from different modes
+//                --> Energy      mode (0)
+//                    Temperature mode (1)
+//                    Entropy     mode (2)
+//                    Pressure    mode (3)
+//
+// Note        :  1. Use 3D Lagrange linear interpolation or Catmull-Rom cubic interpolation formulae
+//                   to search the corresponding temperature/energy given (rho, eps/temp/entr/pres, Ye)
+//                2. Invoked by nuc_eos_C_short()
+//
+// Parameter   :  x             : Input vector of first  variable (rho)
+//                var           : Input vector of second variable (eps/temp/entr/pres)
+//                z             : Input vector of third  variable (Ye)
+//                found_ltoreps : Output log(temp)/log(eps) of interpolated function values
+//                table_main    : The main      tabulated nuclear EoS table
+//                table_Aux     : The auxiliary tabulated nuclear EoS table
+//                nx            : X-dimension of table
+//                ny            : Y-dimension of table
+//                nz            : Z-dimension of table
+//                nvar          : Size of input quantity in the auxiliary table
+//                xt            : Vector of x-coordinates of table
+//                yt            : Vector of y-coordinates of table
+//                zt            : Vector of z-coordinates of table
+//                vart          : Vector of the input quantity
+//                keymode       : Which mode to be used
+//                keyerr        : Output error
+//
+// Return      :  found_ltoreps
+//-------------------------------------------------------------------------------------
+GPU_DEVICE
+void findtoreps_direct( const real x, const real var, const real z,
+                        real *found_ltoreps, const real *table_main, const real *table_aux,
+                        const int nx, const int ny, const int nz, const int nvar,
+                        const real *xt, const real *yt, const real *zt, const real *vart,
+                        const int keymode, int *keyerr )
+{
+
+   int  ix, iy_brd1, iy_brd2, iz, iv, ip, ifs;
```

Comment:
For the variable names with `brd`, I think those are related to `boundary`. Maybe we would better name them with `bdy` or `bdr`.

Reply:
Use `bdry` for abbreviation.